pkg/plugins/label/label.go

```diff
@@ -143,6 +143,23 @@ func getLabelsFromGenericMatches(matches [][]string, labelFilter func(string) bo
 	return labels
 }
 
+// getCategoriesFromLabels extracts the categories from `labels` that are also in `needsLabels`.
+// Example:
+//
+//	labels 		:= []string{"triage/unresolved", "area/testing"}
+//	needsLabels := []string{"triage"}
+//	result 		:= []string{"triage"}
+func getCategoriesFromLabels(labels, needsLabels []string) sets.Set[string] {
```

Comment:
```suggestion
func getCategoriesFromLabels(labels, needsLabels []string) sets.Set[string] {
	if len(labels) == 0 || len(needsLabels) == 0 {
		return sets.New[string]() 
	}
	needsLabelsSet := sets.New(needsLabels...)
	categories := sets.New[string]()
	for _, label := range labels {
		if prefix, _, found := strings.Cut(label, "/"); found && needsLabelsSet.Has(prefix) {
			categories.Insert(prefix)
		}
	}
	return categories
}
```
Just for you to consider. Returns early and more visible split.

Reply:
Weak -1 from me, I'm a fan of private-unless-proven-to-be-necessary-public :sweat:

Reply:
Its my bad. I meant insights of the function, my intention was not to propose to make it public. Let me correct that.

Reply:
Corrected, again comment is about internals not about publicity, sorry for the confusion.